File: android-app/src/main/java/com/medziku/plantwatering/station/PlantStationDetailsFragment.java
package com.medziku.plantwatering.station;

import android.content.Context;
import android.os.Bundle;
import android.os.Parcelable;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;

import com.medziku.plantwatering.MainActivity_;
import com.medziku.plantwatering.R;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.EFragment;
import org.androidannotations.annotations.ViewById;

import java.util.ArrayList;
import java.util.List;

@EFragment(R.layout.plant_station_details_fragment)
public class PlantStationDetailsFragment extends Fragment {

    public static final String BACKSTACK_NAME = "psdf";

    private static final String ARG_PLANT_STATION = "ps";

    @ViewById(R.id.toolbar)
    protected Toolbar toolbar;

    @ViewById(R.id.partsRecyclerView)
    protected RecyclerView recyclerPartsRecyclerView;

    private OnFragmentInteractionListener mListener;
    private PlantStation mPlantStation;

    public PlantStationDetailsFragment() {
        // Required empty public constructor
    }

    public static PlantStationDetailsFragment newInstance(PlantStation station) {
        PlantStationDetailsFragment_ fragment = new PlantStationDetailsFragment_();
        if (station instanceof Parcelable) {
            Bundle args = new Bundle();
            args.putParcelable(ARG_PLANT_STATION, (Parcelable) station);
            fragment.setArguments(args);
        }
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            Parcelable parcelable = getArguments().getParcelable(ARG_PLANT_STATION);
            if (parcelable instanceof PlantStation) {
                mPlantStation = (PlantStation) parcelable;
            }
        }
    }

    @AfterViews
    protected void setUpToolbar() {
        ((MainActivity_) getActivity()).setActionBar(toolbar);
    }

    @AfterViews
    protected void setUpStationPartRecyclerView() {
        List<PlantStationPart> list;
        if (mPlantStation.getPlantStationParts() instanceof List) {
            list = (List) mPlantStation.getPlantStationParts();
        } else {
            list = new ArrayList<>(mPlantStation.getPlantStationParts());
        }
        PlantStationPartsAdapter adapter = new PlantStationPartsAdapter(list);
        recyclerPartsRecyclerView.setAdapter(adapter);
        recyclerPartsRecyclerView.setLayoutManager(new LinearLayoutManager(this.getContext()));

    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnFragmentInteractionListener) {
            mListener = (OnFragmentInteractionListener) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        mListener = null;
        super.onDetach();
    }

    public interface OnFragmentInteractionListener {
    }
}

File: android-app/src/main/kotlin/com/medziku/plantwatering/station/PlantStationDetailsFragment.kt
package com.medziku.plantwatering.station

import android.content.Context
import android.os.Bundle
import android.os.Parcelable
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.Toolbar
import com.medziku.plantwatering.MainActivity_
import com.medziku.plantwatering.R
import org.androidannotations.annotations.AfterViews
import org.androidannotations.annotations.EFragment
import org.androidannotations.annotations.ViewById
import java.util.*

@EFragment(R.layout.plant_station_details_fragment)
open class PlantStationDetailsFragment : Fragment() {

    private var listener: OnFragmentInteractionListener? = null
    private var plantStation: PlantStation? = null

    @ViewById(R.id.toolbar)
    protected lateinit var toolbar: Toolbar

    @ViewById(R.id.partsRecyclerView)
    protected lateinit var recyclerPartsRecyclerView: RecyclerView

    companion object {
        const val ARG_PLANT_STATION = "ps"
        const val BACKSTACK_NAME = "psdf"

        fun newInstance(plantStation: PlantStation): PlantStationDetailsFragment {
            val fragment = PlantStationDetailsFragment_()
            if (plantStation is Parcelable) {
                val args = Bundle()
                args.putParcelable(ARG_PLANT_STATION, plantStation)
                fragment.arguments = args
            }
            return fragment
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            val parcelable: Parcelable? = arguments.getParcelable(ARG_PLANT_STATION)
            if (parcelable is PlantStation) {
                plantStation = parcelable
            }
        }
    }

    @AfterViews
    protected fun setUpToolbar() {
        (activity as MainActivity_).setActionBar(toolbar)
    }

    @AfterViews
    protected fun setUpStationPartRecyclerView() {
        val list: List<PlantStationPart> = if (plantStation?.plantStationParts is List<*>) {
            plantStation?.plantStationParts as List<PlantStationPart>
        } else {
            ArrayList<PlantStationPart>(plantStation?.plantStationParts)
        }
        val adapter = PlantStationPartsAdapter(list)
        recyclerPartsRecyclerView.adapter = adapter
        recyclerPartsRecyclerView.layoutManager = LinearLayoutManager(this.context)
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context is OnFragmentInteractionListener) {
            listener = context
        } else {
            throw RuntimeException("$context must implement OnFragmentInteractionListener")
        }
    }

    override fun onDetach() {
        listener = null
        super.onDetach()
    }

    interface OnFragmentInteractionListener {

    }

}